### ISSUE TITLE
update gradient every N batch

### ISSUE THREAD

**Issue by as754770178** [pytorch/pytorch#24898]
my code:
```
loss.backward()
if iter_num % iter_size == 0:
    optimizer.step()
    optimizer.zero_grad()
```
When `iter_size == 8`, the loss become `NAN`.
(1) I want to know whether the loss is sum, not average?
(2) when optimizer.step(), the gradient is sum or average?

**Comment by gchanan** [comment]
Please use https://discuss.pytorch.org (https://discuss.pytorch.org/) for questions. Please kindly feel encouraged to reopen this issue, if this does not apply.
